### PR TITLE
Fix crash upon adding new nodes to OSS cluster [MOD-5778]

### PR DESCRIPTION
**Pull request by alonre24**
Addressing #3801 #3816

**Describe the changes in the pull request**

Upon adding a node to an existing cluster with `cluster meet` command, a crash occurred in `SetMyPartition` function, since `myshard` property is uninitialized. That property is set upon updating the cluster topology (REFRESH CLUSTER is called periodically every few seconds). However, the newly created shard is not considered part of the topology until it is assigned to hold slots (upon initializing, no slots are assigned). Hence, the property wasn't set, and we got the crash.

This PR is introducing a patch, so we won't call `SetMyPartition` and access `myshard` property if it hasn't been initialized yet.

**Mark if applicable**

- [ ] This PR introduces API changes
- [ ] This PR introduces serialization changes

### CODE REVIEW

**Comment by emberkeith** [comment]
This seemed to fix #3816 for me. Thank you!

File: tests/pytests/test_multithread.py
Comment (DvirDukhan) [inline tests/pytests/test_multithread.py:1]: why the additions in this file?

File: tests/pytests/test_issues.py [hunk 1]
```diff
@@ -818,3 +821,70 @@ def test_mod5791(env):
...
+
+
+@skip(asan=True)
+def test_mod5778_add_new_shard_to_cluster(env):
+    SkipOnNonCluster(env)
+    env.assertEqual(len(env.cmd('CLUSTER SHARDS')), len(env.envRunner.shards))
+
+    # Create a new redis instance with redisearch loaded.
+    # TODO: add appropriate APIs to RLTest to avoid this abstraction braking.
+    new_instance_port = env.envRunner.shards[-1].port + 2  # use a fresh port
```
Comment (oshadmi) [inline tests/pytests/test_issues.py:833]: Can fail on some platforms? (need to ask for an available port?)

**Review by oshadmi — COMMENTED** [review]
👍🏼 Looks good! One comment

**Review by DvirDukhan — APPROVED** [review]

**Pull request merged**